qiskit_quantuminspire/qi_jobs.py

```diff
@@ -35,15 +59,92 @@ def __init__(
         """Submit the (batch)job to the quantum inspire backend.
 
         Use compute-api-client to call the cjm endpoints in the correct order, to submit the jobs.
         """
-        for i in range(1, 3):
-            self._job_ids.append(str(i))
-        self.job_id = "999"  # ID of the submitted batch-job
+        options = self.backend().options
+        configuration = config()
+        # call create algorithm
+        async with ApiClient(configuration) as api_client:
+
+            # always create a list
+            circuits: List[QuantumCircuit] = []
+            if isinstance(self._run_input, QuantumCircuit):
+                circuits.append(self._run_input)
+            else:
+                circuits = self._run_input
+
+            project = await self._create_project(api_client, configuration.auth_settings().team_member_id)
+            batch_job = await self._create_batch_job(api_client, backend_type_id=self.backend().id)
+
+            # iterate over the circuits
+            for circuit in circuits:
+                algorithm = await self._create_algorithm(api_client, project)
+                commit = await self._create_commit(api_client, algorithm)
+                file = await self._create_file(api_client, commit, circuit)
+                job: Job = await self._create_job(
+                    api_client, file, batch_job, number_of_shots=options.get("shots", default=1000)
+                )
+                self._job_ids.append(job.job_id())
+
+            await self._enqueue_batch_job(api_client, batch_job)
+            self.job_id = batch_job.id
+
+    async def _create_project(self, api_client: ApiClient, owner_id: int) -> Project:
+        api_instance = ProjectsApi(api_client)
+        obj = ProjectIn(
+            owner_id=owner_id,
+            name=self.program_name,
+            description="Project created by SDK",
+            starred=False,
+        )
+        return await api_instance.create_project_projects_post(obj)
+
+    async def _create_algorithm(self, api_client: ApiClient, project: Project) -> Algorithm:
+        api_instance = AlgorithmsApi(api_client)
+        obj = AlgorithmIn(
+            project_id=project.id, type=AlgorithmType.QUANTUM, shared=ShareType.PRIVATE, name=self.program_name
+        )
+        return await api_instance.create_algorithm_algorithms_post(obj)
+
+    async def _create_commit(self, api_client: ApiClient, algorithm: Algorithm) -> Commit:
+        api_instance = CommitsApi(api_client)
+        obj = CommitIn(
+            description="Commit created by SDK",
+            algorithm_id=algorithm.id,
+        )
+        return await api_instance.create_commit_commits_post(obj)
+
+    async def _create_file(self, api_client: ApiClient, commit: Commit, circuit: QuantumCircuit) -> File:
+        api_instance = FilesApi(api_client)
+        obj = FileIn(
+            commit_id=commit.id,
+            content=str(circuit),  # TODO: use cQasm here
+            language_id=1,
```

Comment:
Bit of a magic number.

Reply:
This was taken in the same way as in the SDK:  
https://github.com/QuTech-Delft/quantuminspire2/blob/20ac6c37b6e3beab540f5685e718bdc7ce04d97d/quantuminspire/util/api/remote_backend.py#L148

Reply:
In the SDK this is also a magic number, and we have a nice opportunity to fix this. We can't really guarantee this id will always refer to the correct language right?

Reply:
What I do now is call the languages api, and search through it until I find one that matches "cqasm". I am doubting if that should include the version number, but can be easily added.

Reply:
Yeah, I made note about the version number in the _get_language() method.